src/Auth/Storage/StorageInterface.php

```diff
@@ -30,10 +30,10 @@ public function read();
     /**
      * Write user record.
      *
-     * @param array $user User record.
+     * @param array|\ArrayAccess $user User record.
      * @return void
      */
-    public function write(array $user);
+    public function write($user);
```

Comment:
I did [comment](https://github.com/cakephp/cakephp/issues/8240#issuecomment-186300201) about the typehint change (albeit for AuthComponent::setUser()) on the original issue.